sale_order_split_strategy/models/sale_order.py

```diff
@@ -0,0 +1,157 @@
+        return self.order_line.filtered(lambda l: not l._is_delivery()) == lines_to_split
+
+    def _split_lines(self, sections_dict, lines_to_split, target_order):
+        self.ensure_one()
+        copy_sections = self.split_strategy_id.copy_sections
+        copy_notes = self.split_strategy_id.copy_notes
+        for section_id, line_ids in sections_dict.items():
+            section_line = self.env["sale.order.line"].browse(section_id)
+            section_lines = self.env["sale.order.line"].browse(line_ids)
+            for line in section_lines:
+                if line in lines_to_split:
+                    line.write(target_order._prepare_order_split_line_move_defaults())
+                # FIXME: We should consider if the line is part of the section
+                if copy_notes and line.display_type == "line_note":
+                    line.copy(target_order._prepare_order_split_line_move_defaults())
+            lines_in_section_to_split = [
+                li in lines_to_split for li in section_lines if not li.display_type
+            ]
+            # If all lines in section are to split, move the section to split order
+            if copy_sections and section_line and all(lines_in_section_to_split):
+                section_line.write(
+                    target_order._prepare_order_split_line_move_defaults()
+                )
+            # If any line in section is to split, copy the section to split order
+            elif copy_sections and section_line and any(lines_in_section_to_split):
+                section_line.copy(
+                    target_order._prepare_order_split_line_move_defaults()
+                )
+        return True
+
+    def _get_lines_grouped_by_sections(self):
+        # Prepare a dict from sale order lines and their sections
+        self.ensure_one()
+        sections_dict = defaultdict(list)
+        section_id = None
+        for line in self.order_line.sorted():
+            if line.display_type == "line_section":
+                section_id = line.id
+            sections_dict[section_id].append(line.id)
+        return sections_dict
+
+    def _prepare_message_split_from(self, order):
+        self.ensure_one()
+        template = self.env["ir.ui.view"].search(
+            [("key", "=", "sale_order_split_strategy.split_from")], limit=1
+        )
+        if not template:
+            return _(
+                "This sale order was created after splitting lines from %(order_name)s"
+                " using strategy %(strategy)s",
+                order_name=order.name,
+                strategy=order.split_strategy_id.name,
+            )
+        else:
+            return template._render(values={"from_order": order})
+
+    def _prepare_message_split_to(self, order):
+        self.ensure_one()
+        template = self.env["ir.ui.view"].search(
+            [("key", "=", "sale_order_split_strategy.split_to")], limit=1
+        )
+        if not template:
+            return _(
```

Comment:
```suggestion
        if template:
            return template._render(values={"from_order": self, "to_order": order})
        return _(...)
```

Reply:
Actually we need to use `_render_template` because standard user don't have access to `ir.ui.view` to search